### PR TITLE
feat: Support snapshot queries

### PR DESCRIPTION
**Pull request by taka-oyama**
Fix: #145 

## Checklist

- [x] CHANGELOG

## Summary by CodeRabbit

- **New Features**
  - Introduced support for snapshot queries, allowing users to retrieve historical data.
  - Added methods for managing and checking snapshot usage within queries.
- **Improvements**
  - Enhanced query execution performance by automatically wrapping values in `UNNEST` for large datasets.
  - Improved error handling in transaction processes during snapshot states.

### CODE REVIEW

**Comment by taka-oyama** [comment]
@coderabbitai review

**Comment by matthewjumpsoffbuildings** [comment]
@taka-oyama anything I can do to help this PR be ready for merge?

**Comment by taka-oyama** [comment]
Unfortunately, I'm going to be on paternity leave for about a month, so I won't be able to work on this until august.

The reason there is no progress here is because I tried to use SingleUse Transaction to implement this, but there seems to be a bug in `google/cloud-spanner` which causes all queries running as SingleUse to throws an error. This needs to be resolved first and I haven't gotten around to creating a issue in https://github.com/googleapis/google-cloud-php.

**Comment by matthewjumpsoffbuildings** [comment]
> there seems to be a bug in `google/cloud-spanner` which causes all queries running as SingleUse to throws an error.

Is this just in the emulator? Or on real Spanner instances too?

**Comment by matthewjumpsoffbuildings** [comment]
@taka-oyama any updates on this PR?

File: src/Connection.php [hunk 1]
```diff
@@ -33,13 +33,15 @@
 use Google\Cloud\Core\Exception\NotFoundException;
 use Google\Cloud\Spanner\Database;
 use Google\Cloud\Spanner\Session\SessionPoolInterface;
+use Google\Cloud\Spanner\Snapshot;
```
Comment (halnique) [inline src/Connection.php:36]: nits: unused
```suggestion
```

**Review by halnique — APPROVED** [review]
Almost LGTM.

**Pull request merged**